subsys/dfu/img_util/flash_img.c

```diff
@@ -43,44 +43,75 @@ BUILD_ASSERT((CONFIG_IMG_BLOCK_BUF_SIZE % FLASH_WRITE_BLOCK_SIZE == 0),
 	     "FLASH_WRITE_BLOCK_SIZE");
 #endif
 
+static int scramble_mcuboot_trailer(struct flash_img_context *ctx)
+{
+	int rc = 0;
+
+#ifdef CONFIG_IMG_ERASE_PROGRESSIVELY
+	if (stream_flash_bytes_written(&ctx->stream) == 0) {
+		off_t toff = boot_get_trailer_status_offset(ctx->flash_area->fa_size);
+		off_t offset;
+		size_t size;
+		const struct flash_parameters *fparams =
+				flash_get_parameters(flash_area_get_device(ctx->flash_area));
+#ifdef CONFIG_STREAM_FLASH_ERASE
+		/* for erasable devices prgressive-erase works only along with CONFIG_STREAM_FLASH_ERASE option*/
+		if (flash_params_get_erase_cap(fparams) & FLASH_ERASE_C_EXPLICIT) {
+			/* On devices with explicit erase we are aligning to page
+			 * layout.
+			 */
+			struct flash_pages_info info;
+
+			rc = flash_get_page_info_by_offs(flash_area_get_device(ctx->flash_area), toff,
+							 &info);
+			if (rc != 0) {
+				return rc;
+			}
+			offset = info.start_offset;
+			size = info.size;
+
+		} else
+#endif
+		{
+
+			/* On devices with no erase, we are aligning to write block
```

Comment:
Since now, the `CONFIG_IMG_ERASE_PROGRESSIVELY` Kconfig option is supported for devices with no erase. Could you also align this Kconfig dependency and remove `depends on FLASH_HAS_EXPLICIT_ERASE`?

```
config IMG_ERASE_PROGRESSIVELY
	bool "Erase flash progressively when receiving new firmware"
	select STREAM_FLASH_ERASE
	depends on FLASH_HAS_EXPLICIT_ERASE
```

->

```
config IMG_ERASE_PROGRESSIVELY
	bool "Erase flash progressively when receiving new firmware"
	select STREAM_FLASH_ERASE if FLASH_HAS_EXPLICIT_ERASE
```